ipatests/test_integration/test_host.py

```diff
@@ -0,0 +1,109 @@
+        """
+        Several tests of ipa host-mod and ipa host-add with defferent options,
+        working with macaddress and ip address.
+        """
+
+        host = self.master
+        # chosen random MAC address for test
+        host_macaddr = u'00:50:56:30:F6:5F'
+        tasks.kinit_admin(host)
+
+        try:
+            # addattr with option macaddress
+            host.run_command(["ipa host-mod --addattr macaddress=",
+                              host_macaddr, host.hostname])
+
+            self.check_host_contains_attribute(host, host_macaddr)
+
+            # delattr with option macaddress with incorrect value
+            incorrect_host_macaddress = u'00:50:56:30:F6:EE'
+            result_incorrect_mac = host.run_command(
+                ["ipa host-mod --delattr macaddress=",
+                 incorrect_host_macaddress, host],
+                raiseonerr=False
+            )
+            assert result_incorrect_mac.returncode != 0
+
+            # delattr with option macaddress
+            host.run_command(["ipa host-mod --delattr macaddress=",
+                              host_macaddr, host])
+
+            host_show_result = host.run_command(["ipa host-show", host,
+                                                 "--all"])
+            assert host_macaddr not in host_show_result.stdout_text
+            ldapsearch_result = host.run_command(["ldapsearch -h", host])
+            assert host_macaddr not in ldapsearch_result.stdout_text
+
+            # setattr with option macaddress
+            host.run_command(["ipa host-mod --setattr macaddress=",
+                              host_macaddr, host])
+
+            self.check_host_contains_attribute(host, host_macaddr)
+
+            # setattr with option macaddress and addattr on macaddress
+            # shouldn't be multivalue - additional add should fail
+            result = host.run_command(["ipa host-mod --addattr macaddress=",
+                                       host_macaddr, host])
+            assert result.returncode != 0
+
+            # delattr with option macaddress with lowercase
+            host_macaddr_lowercase = host_macaddr.lower()
+            host.run_command(["ipa host-mod --delattr macaddress=",
+                              host_macaddr_lowercase, host])
+
+            # add a host with options macaddress and force
+            host.run_command(["ipa host-add", host, "--macaddress=",
+                              host_macaddr, "--force"])
+
+            self.check_host_contains_attribute(host, host_macaddr)
+
+            # hostmod of a host with option macaddress
+            modified_macaddr = ":".join(host_macaddr.split(":")[:-1]) + ":EE"
+            host.run_command(["ipa host-mod --macaddress=",
+                              modified_macaddr, host])
+
+            self.check_host_contains_attribute(host, modified_macaddr)
+        finally:
+            host.run_command(["ipa host-del", host.hostname])
```

Comment:
If the test fails, it will delete the master entry and break everything. Instead of playing with the master's entry, the test can create a fake host with ipa host-add testhost.<domain> --force and use this testhost throughout the test.